.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,64 @@
+name: SDM Docker Build
+
+on:
+  push:
+    branches:
+      - main
+      - christo/sdm-test # (dev/test branch)
+    paths:
+      - 'airbyte_cdk/**'
+      - '.github/workflows/docker-build.yml'
+      - 'Dockerfile'
+  workflow_dispatch:
+    inputs:
+      version-tag:
+        description: "Version tag for the image (optional)"
+        required: false
+        type: string
```

Comment:
My understanding is that we'll publish `latest` on each cdk change, without pushing a second tag with the actual version. Actual version tags will be published manually with workflow dispatch. 
Is it what you want?
My concern is that we'll publish SDM image without an explicit CDK version to use. 
It'll use whatever version exists on the branch at the time the workflow is triggered. It breaks build reproducibility. 
We'd not be able to re-publish SDM with a different CDK version unless we revert commits in the repo.
But reverting commits could mean reverting GHA changes too.
And SDM version can't be coupled to CDK version (we want to be able to change the SDM base image without publishing a new CDK version.)

I'd suggest:
* Version controlling which CDK version SDM gets
* Installing this version in SDM from Pypi